addon/components/json-viewer.js

```diff
@@ -40,43 +39,37 @@ function getPath(node) {
   }
 }
 
-export default Component.extend({
-  classNames: ['json-viewer'],
-  layout,
-
-  // passed-in
-  json: null,
-
-  displayOptions: computed('options', function () {
-    let options = this.options || {};
+export default class JsonViewer extends Component {
+  get displayOptions() {
+    let options = this.args.options || {};
     assert(
       `Only allowed options are: ${ALLOWED_OPTIONS}`,
       Object.keys(options).every((key) => ALLOWED_OPTIONS.includes(key)),
     );
     return options;
-  }),
+  }
 
-  didInsertElement() {
-    this._super(...arguments);
-    this._copyHandler = (evt) => {
-      let [startNode, startOffset, endNode, endOffset] = getOrderedSelection();
+  @action copyListener(evt) {
+    let [startNode, startOffset, endNode, endOffset] = getOrderedSelection();
 
-      let startPath = getPath(startNode);
-      let endPath = getPath(endNode);
-      let range = {
-        start: { path: startPath, index: startOffset },
-        end: { path: endPath, index: endOffset },
-      };
-      let str = jsonStringify(this.json, range);
-      evt.clipboardData.setData('text/plain', str);
-      evt.preventDefault();
+    let startPath = getPath(startNode);
+    let endPath = getPath(endNode);
+    let range = {
+      start: { path: startPath, index: startOffset },
+      end: { path: endPath, index: endOffset },
     };
-    this.element.addEventListener('copy', this._copyHandler);
-  },
+    let str = jsonStringify(this.args.json, range);
+    evt.clipboardData.setData('text/plain', str);
+    evt.preventDefault();
+  }
 
-  willDestroyElement() {
-    this._super(...arguments);
-    this.element.removeEventListener('copy', this._copyHandler);
-    this._copyHandler = null;
-  },
-});
+  @action
+  addCopyListener(element) {
+    element.addEventListener('copy', this.copyListener);
```

Comment:
Why not `{{on "copy" this.copyListener}}` in the template? https://api.emberjs.com/ember/3.28/classes/Ember.Templates.helpers/methods/on?anchor=on

Reply:
done in ecdee5526604091ca3b13fba472050b14840b6d0

This also allows dropping the `@ember/render-modifiers` dep, thanks for making this suggestion